packages/core/utils/src/product/csv-normalizer.ts

```diff
@@ -0,0 +1,518 @@
+      (token) => token.startsWith("[") && token.endsWith("]")
+    )
+
+    if (!iso) {
+      throw createError(
+        rowNumber,
+        `Invalid price format used by "${columnName}". Expect column name to contain the ISO code inside square brackets. For example: "Variant Price [USD]"`
+      )
+    }
+
+    return {
+      iso: iso.replace(/^\[|\]$/g, "")
+        .toLowerCase(),
+      region: priceTokens.find((token) => token !== iso),
+    }
+  }
+
+  throw createError(
+    rowNumber,
+    `Invalid price format used by "${columnName}". Expect column name to contain the ISO code. For example: "Variant Price USD"`
+  )
+}
+
+/**
+ * Processes a column value as a string
+ */
+function processAsString<Output>(
+  inputKey: string,
+  outputKey: keyof Output
+): ColumnProcessor<Output> {
+  return (csvRow, _, __, output) => {
+    const value = normalizeValue(csvRow[inputKey])
+    if (isPresent(value)) {
+      output[outputKey as any] = value
+    }
+  }
+}
+
+/**
+ * Processes the column value as a string
+ */
+function processAsBoolean<Output>(
+  inputKey: string,
+  outputKey: keyof Output
+): ColumnProcessor<Output> {
+  return (csvRow, _, __, output) => {
+    const value = normalizeValue(csvRow[inputKey])
+    if (isPresent(value)) {
+      output[outputKey as any] = tryConvertToBoolean(value, value)
+    }
+  }
+}
+
+/**
+ * Processes the column value as a number
+ */
+function processAsNumber<Output>(
+  inputKey: string,
+  outputKey: keyof Output,
+  options?: { asNumericString: boolean }
+): ColumnProcessor<Output> {
+  return (csvRow, _, rowNumber, output) => {
+    const value = normalizeValue(csvRow[inputKey])
+    if (isPresent(value)) {
+      const numericValue = tryConvertToNumber(value)
+      if (numericValue === undefined) {
+        throw createError(
+          rowNumber,
+          `Invalid value provided for "${inputKey}". Expected value to be a number, received "${value}"`
+        )
+      } else {
+        if (options?.asNumericString) {
+          output[outputKey as any] = String(numericValue)
+        } else {
+          output[outputKey as any] = numericValue
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Processes the CSV column as a counter value. The counter values
+ * are defined as "<Column Name> <1>". Duplicate values are not
+ * added twice.
+ */
+function processAsCounterValue<Output extends Record<string, any[]>>(
+  inputMatcher: RegExp,
+  arrayItemKey: string,
+  outputKey: keyof Output
+): ColumnProcessor<Output> {
+  return (csvRow, rowColumns, _, output) => {
+    output[outputKey] = output[outputKey] ?? []
+    const existingIds = output[outputKey].map((item) => item[arrayItemKey])
+
+    rowColumns
+      .filter((rowKey) => inputMatcher.test(rowKey))
+      .forEach((rowKey) => {
+        const value = normalizeValue(csvRow[rowKey])
+        if (!existingIds.includes(value) && isPresent(value)) {
+          output[outputKey].push({ [arrayItemKey]: value })
+        }
+      })
```

Comment:
It is more of a suggestion but as you prefer

```suggestion
    const existingIds = new Set(output[outputKey].map((item) => item[arrayItemKey]))

    rowColumns
      .filter((rowKey) => inputMatcher.test(rowKey))
      .forEach((rowKey) => {
        const value = normalizeValue(csvRow[rowKey])
        if (isPresent(value) && !existingIds.has(value)) {
          output[outputKey].push({ [arrayItemKey]: value })
        }
      })
```